internal/pkg/cli/task_run.go

```diff
@@ -253,8 +343,8 @@ func (o *runTaskOpts) askEnvName() error {
 		return nil
 	}
 
-	// NOTE: if we are not in any workspace and app flag is not specified, use the "None" environment.
-	if  o.AppName() == "" {
+	// NOTE: if the subnets are not provided, we are not in any workspace and app flag is not specified, use the "None" environment.
```

Comment:
Now that I think about it :P, I don't think I understand why we return nil when `o.AppName()` is empty. Doesn't that mean that they're not in a Copilot workspace and we should be prompting them so that they can choose either a Copilot env or the default vpc?

Reply:
That makes sense. I thought if they want to deploy the tasks to an app, they would have provided the app flag --- but yes, it's weird that we prompt for env if env is not provided, but are requiring the user to specify an app if they want to use an env.

Reply:
In fact, I think it's probably better if I make this change in a separate PR. Since we want to provide an additional option (similar to None in env), I need to refactor selector.